Draft-Accepted/RFC0066-PowerShell-User-Content-Location.md

```diff
@@ -0,0 +1,133 @@
+- A configuration file in the PowerShell user content folder will determine the location of the user
+  scoped **PSModulePath**.
+  - By default, the PowerShell user content folder will be located in the
+    `$env:LOCALAPPDATA\PowerShell`.
+  - The new location becomes the location used as the `CurrentUser` scope for PSResourceGet.
+- The proposed directory structure:
+
+  ```
+  C:\Users\UserName\AppData\Local\PowerShell\
+  ├── powershell.config.json   (Not Configurable)
+  ├── Scripts                  (Configurable)
+  ├── Modules                  (Configurable)
+  ├── Help                     (Configurable)
+  └── <*profile>.ps1           (Configurable)
+  ```
+
+- The following setting is added to the `powershell.config.json` file:
+
+  **UserPSContentPath** specifies the full path of the content folder. The default value is
+  `$env:LOCALAPPDATA\PowerShell`. The user can change this value to a different path.
+
+  ```json
+  {
+      "UserPSContentPath" : "$env:LOCALAPPDATA\\PowerShell",
+  }
+  ```
+
+## User Experience
+
+- On startup PowerShell will create a directory in AppData and a configuration file.
+- The user scoped **PSModulePath** will point to `Modules` folder under the location specified by
+  **UserPSContentPath**.
+- Users can configure a custom location for PowerShell user content by changing the value of
+  **UserPSContentPath**.
+- Users will initially need to manually move/copy their existing PowerShell user content 
+  from the Documents folder to the new location after enabling the feature.
+- We intend to add a new cmdlet to aid this process in a future release shortly 
+  after this work has been completed as part of the work into hopefully the 7.6.0 GA
+  release or another 7.6.x release. Details on this to follow.
+
+## Other considerations
+
+- The following functionalities will be affected:
+  - SecretManagement
+    - SecretManagement extension vaults are registered for the current user context in:
+      `$env:LOCALAPPDATA\Microsoft\PowerShell\secretmanagement\secretvaultregistry\`
+
+      When an extension vault is registered, SecretManagement stores the full path to the extension
+      module in the registry. Moving the PowerShell content to a new location will break the vault
+      registrations.
+  - Document instructions on how to re-register vaults after moving the content folder.
+  - Document the need to keep Modules in the Documents folder to so that SecretManagement
+    continues to work for multiple installs of PowerShell 7 (stable and preview).
+
+- Use the following script to copy the PowerShell contents folder:
+
+  ```pwsh
+  $newPath = "C:\Custom\PowerShell\Modules"
+  $currentUserModulePath = [System.Environment]::GetFolderPath('MyDocuments') + "\PowerShell"
+  Copy-Item -Path $currentUserModulePath -Destination $newPath -Recurse -Force
+  ```
+
+- PowerShellGet is hardcoded to install scripts and modules in the user's `Documents` folder. It
+  will not support this feature.
```

Comment:
We may need to recommend regualar checks for resources installed via PowerShellGet 

However it also would be a good opportunity for us to actually look at the removal of PowerShellGet & PackageManagement (the sooner the better IMO!) 
This however is sadly really a question that is out of scope for this RFC but closely aligned.